src/components/CustomerQuota/CustomerQuotaScreen.tsx

```diff
@@ -25,126 +34,270 @@ const styles = StyleSheet.create({
+                  onToggle={() =>
+                    setCart(cart => ({
+                      ...cart,
+                      [category]: !cart[category]
+                    }))
+                  }
+                />
+              </View>
+            );
+          })}
+      </View>
+    </CustomerCard>
+    <View style={[styles.ctaButtonsWrapper, styles.buttonRow]}>
+      <View
+        style={[styles.submitButton, !isLoading && { marginRight: size(2) }]}
+      >
         <DarkButton
-          text="Buy 1 box"
+          text="Checkout"
+          icon={
+            <Feather
+              name="shopping-cart"
+              size={size(2)}
+              color={color("grey", 0)}
+            />
+          }
           onPress={onRecordTransaction}
           isLoading={isLoading}
+          fullWidth={true}
         />
       </View>
-      {!isLoading && <SecondaryButton text="Cancel" onPress={onCancel} />}
+      {!isLoading && (
+        <SecondaryButton
+          text="Cancel"
+          onPress={() => {
+            Alert.alert("Cancel transaction?", undefined, [
+              {
+                text: "No"
+              },
+              {
+                text: "Yes",
+                onPress: onCancel,
+                style: "destructive"
+              }
+            ]);
+          }}
+        />
+      )}
     </View>
-  </>
+  </View>
 );
 
-const CannotBuyResult: FunctionComponent<{ onCancel: () => void }> = ({
-  onCancel
-}) => (
-  <>
-    <AppText style={styles.emoji}>❌</AppText>
-    <AppText style={styles.statusTitle}>Customer cannot purchase</AppText>
-    <AppText style={styles.statusMessage}>
-      Customer can purchase 1 box in a week.
-    </AppText>
-    <DarkButton text="Next customer" onPress={onCancel} fullWidth={true} />
-  </>
+const CannotBuyResult: FunctionComponent<{
+  nric: string;
+  onCancel: () => void;
+}> = ({ nric, onCancel }) => (
+  <View>
+    <CustomerCard nric={nric} headerBackgroundColor={color("red", 60)}>
+      <View style={[styles.resultWrapper, styles.failureResultWrapper]}>
+        <AppText style={styles.emoji}>❌</AppText>
+        <AppText style={styles.statusTitle}>Customer cannot purchase</AppText>
+        <AppText>Customer has already used up their quota.</AppText>
+      </View>
+    </CustomerCard>
+    <View style={styles.ctaButtonsWrapper}>
+      <DarkButton text="Next customer" onPress={onCancel} fullWidth={true} />
+    </View>
+  </View>
 );
 
 export const CustomerQuotaScreen: FunctionComponent<NavigationProps> = ({
   navigation
 }) => {
   const { authKey } = useAuthenticationContext();
-  const quota: QuotaResponse = navigation.getParam("quota");
+  const quota: Quota = navigation.getParam("quota");
   const nric: string = navigation.getParam("nric");
-  const [quantity] = useState("1");
+
+  const initialQuantities: CartState = mapValues(
+    keyBy(quota.remainingQuota, "category"),
+    lineQuota => (lineQuota.quantity > 0 ? true : null)
+  );
```

Comment:
Hmm, I'm wondering if we can simplify this since `quota.remainingQuota` is an array of `Transaction`. Thinking just a simple `reduce` would work.

Reply:
fixed